src/datasets/transformations.jl

```diff
@@ -157,6 +161,33 @@ getobs(jdata, 15) == 15
 """
 joinobs(datas...) = JoinedData(datas)
 
+"""
+    tokenize(type, input)
+
+Tokenizes an input string or stream into pieces depending on selected type. 
+"""
```

Comment:
Is there any advantage to this over the default `WordTokenizers.tokenize`? I think it's fine to have a separate path for characters, but for words we'd prefer to rely on their implementation. One way to accomodate this may be to point users towards the experimental [split API](https://github.com/JuliaText/WordTokenizers.jl#experimental-api) and simply define an function overload which uses your character tokenizer.

Reply:
The advantage was that I can control the parameters, so that if WordTokenizers is updated we're not pointing to a function that doesn't exist anymore. I don't quite understand what you mean by the last sentence. Are you suggesting to not implement the tokenizer, and instead tell the user to use the WordTokenizer module's split functions, to implement the same thing with WordTokenizer.tokenize() in place of the case for ``:words``, or something else? The confusion is what you mean by "point".

Reply:
That's correct. We can always re-export `tokenize` from `WordTokenizers` if need be. I mention the split API because it uses multiple dispatch to select which tokenizer to use and thus doesn't require a conditional over a symbol parameter like `:words` or `:chars`. This allows one to swap out tokenizer functions (see https://github.com/JuliaText/WordTokenizers.jl/blob/master/src/split_api.jl for the built-in list), so doing the equivalent of `:chars` could be as easy as calling `collect` on the input string.